chewie/chewie.py

```diff
@@ -283,15 +320,34 @@ def receive_eap_messages(self):
                     packed_message,
                     exception)
                 continue
+
             self.logger.info("Received eap message: %s", str(eap))
             self.send_eap_to_state_machine(eap, dst_mac)
 
+    def receive_ip_activity_messages(self):
+        """Receive IP activity over the 'eap interface' allowing for MAB."""
+        while self.running():
+            sleep(0)
+            self.logger.info("waiting for ip activity.")
+            packed_message = self.ip_activity_socket.receive()
+            self.logger.info("Received ip activity packed_message: %s", str(packed_message))
+            self.send_eth_to_state_machine(packed_message)
+
     def send_eap_to_state_machine(self, eap, dst_mac):
         """sends an eap message to the state machine"""
         self.logger.info("eap EAP(): %s", eap)
         message_id = getattr(eap, 'message_id', -1)
         state_machine = self.get_state_machine(eap.src_mac, dst_mac, message_id)
-        event = EventMessageReceived(eap, dst_mac)
+
+        # Check for response to preemptive_eap
+        preemptive_eap_message_id = self.port_to_eapol_id.get(str(dst_mac), -2)
```

Comment:
In search of a better place to put it. Functionality is moved here and new event type to stop reaching in to 'implementation details' of state machines.